regparser/tree/xml_parser/paragraph_processor.py

```diff
@@ -83,7 +83,12 @@ def separate_intro(self, nodes):
             and labels[0] == mtypes.MARKERLESS
             and labels[1] != mtypes.MARKERLESS)
 
-        if only_one or switches_after_first:
+        def table_first(nodes):
+            if nodes and getattr(nodes[0].source_xml, "tag", "") == "GPOTABLE":
+                return True
+            return False
+
+        if not table_first(nodes) and any([only_one, switches_after_first]):
```

Comment:
Do you think this is any clearer?
```python
first_source = nodes and nodes[0].source_xml
table_first = first_source and first_source.tag == 'GPOTABLE'
if not table_first and any([only_one, switches_after_first]):
```

Inline functions are a bit harder to read, I think